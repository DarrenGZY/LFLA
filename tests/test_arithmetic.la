function main()
{
    var a = 3.5;
    vector b = [1,2,3];
    vector c1 = b *. a;
    vector c2 = b /. a;
    print(c1);
    print(c2);

    matrix d = [1,2,3;6,7,8;];
    matrix e1 = d *. a;
    matrix e2 = d /. a;
    print(e1);
    print(e2);

    matrix m = [3,3;4,2;5,6;];
    matrix n = m * d;
    print(size(m));
    print(size(d));
    print(n);

}
