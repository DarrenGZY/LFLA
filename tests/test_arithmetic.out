[  3.5   7.   10.5]
[ 0.28571429  0.57142857  0.85714286]
[[  3.5   7.   10.5]
 [ 21.   24.5  28. ]]
[[ 0.28571429  0.57142857  0.85714286]
 [ 1.71428571  2.          2.28571429]]
(3, 2)
(2, 3)
[[21 27 33]
 [16 22 28]
 [41 52 63]]
